cpp/CMakeLists.txt

```diff
@@ -358,7 +382,7 @@ add_metadata_gen_target (
   ${TEST_METADATA_TARGET}
   "${RESOURCES_DIR}/PhoneNumberMetadataForTesting.xml"
   "test_metadata"
-  "test_metadata"
+  "metadata"
```

Comment:
this rename conflicts with the metadata.h generated above in the non-test METADATA_TARGET. they both output a `metadata.h` like this, which means the build has two rules that generate the same output file (one for metadata_target, and one for test_metadata_target).

this breaks the build for cmake -G Ninja, because the .ninja spec forbids two distinct rules for the same output:  

```console
$ ninja -C build/
ninja: entering directory 'build/'
ninja: multiple rules generate '/home/demon/src/aports/community/libphonenumber/src/libphonenumber-8.13.4/cpp/src/phonenumbers/metadata.h'
```

Reply:
@nekopsykose, thanks for the catch. The bug was fixed [here](https://github.com/google/libphonenumber/commit/2369340a05076ef235be1550f202ff5b66db9917), but was reintroduced by an clumsy merge. @penmetsaa, will send in a quick pull to restore the fix if that's OK.

Reply:
Regression fixed with [2874](https://github.com/google/libphonenumber/pull/2874).